### PR TITLE
BUG: add names parameter to read_excel

### LINKED ISSUE #12870: read_excel with names keyword argument not working
With pandas 0.18,  pd.read_excel() with name keyworkd does not work anymore.  (pandas 0.17 works fine.) pandas 0.18 has a change to add names as a keyword argument in excel.py at line 76.

#### Code Sample, a copy-pastable example if possible
filename = 'test.xlsx'
column_names = ['field1', 'field2']
df = pd.read_excel(filename, index_col=None,  names=column_names)

#### Expected Output
column_names associated with keyword argument names is supposed to pass down to kwds at C:\Python27\Lib\site-packages\pandas\io\excel.py:177.

kwds is supposed to carry the arguments from application (in this case, column_names above) but it does not get passed. 

#### output of ``pd.show_versions()``
INSTALLED VERSIONS
------------------
commit: None
python: 2.7.11.final.0
python-bits: 32
OS: Windows
OS-release: 7
machine: AMD64
processor: Intel64 Family 6 Model 60 Stepping 3, GenuineIntel
byteorder: little
LC_ALL: None
LANG: None

pandas: 0.18.0
nose: None
pip: 8.1.1
setuptools: 18.3.2
Cython: 0.23.5
numpy: 1.11.0
scipy: None
statsmodels: None
xarray: None
IPython: 4.0.0
sphinx: None
patsy: None
dateutil: 2.5.2
pytz: 2016.3
blosc: None
bottleneck: None
tables: None
numexpr: 2.5
matplotlib: 1.5.1
openpyxl: 2.2.2
xlrd: 0.9.3
xlwt: 1.0.0
xlsxwriter: 0.7.3
lxml: None
bs4: 4.3.2
html5lib: None
httplib2: None
apiclient: None
sqlalchemy: 1.0.12
pymysql: None
psycopg2: None
jinja2: 2.8
boto: None

### PR DESCRIPTION
**Pull request by OXPHOS**
- [x] closes #12870, xref #11874
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

File: pandas/io/tests/test_excel.py [hunk 1]
```diff
@@ -455,6 +455,14 @@ def test_read_one_empty_col_with_header(self):
         expected_header_zero = DataFrame(columns=[0], dtype='int64')
         tm.assert_frame_equal(actual_header_zero, expected_header_zero)
 
+    def test_set_column_names_in_parameter(self):
+        excel = self.get_excelfile('testcolnameset')
```
Comment (jreback) [inline pandas/io/tests/test_excel.py]: can you add the issue number here as a comment

File: pandas/io/tests/test_excel.py [hunk 2]
```diff
@@ -455,6 +455,14 @@ def test_read_one_empty_col_with_header(self):
...
+        refdf = pd.DataFrame([[1, 'foo'], [2, 'bar'],
+                              [3, 'baz']], columns=['A', 'B'])
+
```
Comment (jreback) [inline pandas/io/tests/test_excel.py]: I don't think its necessary to include the files, instead I'd like to test with a dataframe that you write out, these cases:

- write with no header, read back with the specified names (like you are testing here)
- write with a header, and use names (I *think* we replace with the names at this point)

**Comment by OXPHOS** [comment]
Currently, set argument `index_col` = `none` can disable taking column names from excel file.
I can also refactor `excel.py` so that old column names can be automatically dropped when `names=names`

**Example**
If we write `DataFrame([[1, 'foo'], [2, 'bar'], [3, 'baz']], columns=['A', 'B'])` to excel and then read,
```
[In] : read_excel(reader, 'DataFrame', names=['A', 'B'])
[out]: 
   A    B
0  a    b
1  1  foo
2  2  bar
3  3  baz
```

```
[In] : read_excel(reader, 'DataFrame', index_col=None, names=['A', 'B'])
[out]: 
   A    B
0  1  foo
1  2  bar
2  3  baz
```

**Comment by OXPHOS** [comment]
Will remove test files

**Comment by jreback** [comment]
this PR is about a single issue - don't add other things

**Comment by OXPHOS** [comment]
Hey may I ask what is going on with codecov/patch? It pops up some files I didn't touch at all.

**Comment by jreback** [comment]
yeah, code cov was set a bit high.

thanks for the PR!

**Pull request closed without merging**